src/sim.rs

```diff
@@ -660,6 +683,52 @@ mod test {
 
         Ok(())
     }
+    #[test]
+    fn partition_peers_oneway() -> Result {
+        let global = Duration::from_millis(2);
+
+        let mut sim = Builder::new()
+            .min_message_latency(global)
+            .max_message_latency(global)
+            .build();
+
+        // Verify that a node called `sender` can still send to `receiver`,
+        // but not the other way around, after `sim.partition_oneway("receiver", "sender")`;
+
+        sim.host("receiver", || async {
+            let udp_socket = UdpSocket::bind((IpAddr::V4(Ipv4Addr::UNSPECIFIED), 1234)).await?;
+            udp_socket
+                .send_to(&[42], "sender:1234")
+                .await
+                .expect("sending packet should appear to work, even if partitioned");
+            let mut buf = [0];
+            tokio::time::timeout(Duration::from_secs(1), udp_socket.recv_from(&mut buf))
+                .await
+                .expect("receiver should receive a message before the timeout")
+                .expect("the message should be received successfully");
+            Ok(())
+        });
+
+        sim.client("sender", async move {
+            // Peers are partitioned. TCP setup should fail.
```

Comment:
TCP?

Reply:
Ooops😬! A bit of copy-paste may have happened here.

Reply:
Reference to TCP removed!